controller_manager/src/ros2_control_node.cpp

```diff
@@ -40,21 +40,21 @@ int main(int argc, char ** argv)
   std::thread cm_thread([cm]() {
     RCLCPP_INFO(cm->get_logger(), "update rate is %d Hz", cm->get_update_rate());
 
-    std::chrono::system_clock::time_point timepoint_start = std::chrono::system_clock::now();
+    std::chrono::system_clock::time_point timepoint_start(std::chrono::nanoseconds(cm->now().nanoseconds()));
     std::chrono::system_clock::time_point begin = timepoint_start;
     // Use nanoseconds to avoid chrono's rounding
     std::this_thread::sleep_for(std::chrono::nanoseconds(1000000000 / cm->get_update_rate()));
     while (rclcpp::ok())
     {
       std::chrono::system_clock::time_point begin_last = begin;
-      begin = std::chrono::system_clock::now();
+      begin = std::chrono::system_clock::time_point(std::chrono::nanoseconds(cm->now().nanoseconds()));;
       cm->read();
       cm->update(
         rclcpp::Time(
-          std::chrono::duration_cast<std::chrono::nanoseconds>(begin - timepoint_start).count()),
+          std::chrono::duration_cast<std::chrono::nanoseconds>(begin.time_since_epoch()).count()),
```

Comment:
Should/can we use here simply:
```suggestion
          cm->now(),
```

Reply:
It's not exactly the same as what you suggested, but I was able to significantly simpify the PR by storing the update timepoints as `rclcpp::Time` instead of `std::chrono::duration`.